File: guessing_game.py
"""
The following program makes use of a function called the guessing game

"""
import random #importing the random  module 
random.seed(70)  # setting the seed  for the random seed function
random_int = random.randint(1,100)  #setting the range forthe 
#print(random_int) #To know what number the is being guessed. For testing purposes

def guess_game(seed):
    """This is the function for the guessing game"""
    print('Welcome to the Guessing Game!')
    user_guess = int(input("I'm thinking of a number between 1 and 100. Can you guess it?\nGuess:"))#takes user input

    count = 1  #initializing the accumulator variable 
    
    while user_guess != seed: #while loop designed to stop when the user guess is right
        
        count += 1   #increment of the accumulator variable
        
        if user_guess < seed: #checks by how much the user input is more or less than the number
            user_guess = int(input("Good try,but that's too low. Try again.\nGuess:"))
            
        else:
            user_guess = int(input("Good try,but that's too high. Try again.\nGuess:"))
        
    print(f"Yes! You guessed correctly after {count} tries! Congratulations.")#output 
    
    

guess_game(random_int)

File: GuessingGame.java
import java.util.Scanner;
import java.util.Random;     //importing random and scanner module
public class GuessingGame {
    public static void main(String[] args) {
        Random random = new Random(5); //creating an instance of the random object
        int randomint = random.nextInt(1, 101); //initializing the random integer
        //System.out.println(randomint);// printing the random value for testing purposes
        guessGame(randomint);
    }
    public static void guessGame(long seed) { //This is the method for the guessing game
        Scanner keyboard = new Scanner(System.in); // creates  an instance of the scanner object
        System.out.println("Welcome to the Guessing Game!");  //prompts for the user
        System.out.println("I'm thinking of a number between 1 and 100. Can you guess it?");

        int count = 0; //initializing the accumulator variable
        int guess = 0; //intializing the guess variable

        while (guess != seed) {//while loop that continues to run till user guess is true
            System.out.print("Guess:");
            guess = keyboard.nextInt();
            count++;

            if (guess < seed) { //conditional to check the weight of variance of the user input
                System.out.println("Good try, but that's too low. Try again");

            } else if(guess > seed)
            System.out.println("Good try, but that's too high . Try again");

        }
        System.out.println("Yes! You guessed correctly after " + count + " tries! congratulations.");
        keyboard.close();
    }



}









